[Phan Dinh Hung] iP

PDHung1104 opened the pull request.

PDHung1104 committed:
Rename, Greet, Exit

PDHung1104 committed:
Echo

PDHung1104 committed:
Add, List

PDHung1104 committed:
Mark as done

ZhangWenyue3325 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,20 @@
+        String name = "Stella";
+        System.out.println("Hello! I'm " + name);
+        System.out.println("What can I do for you?");
+        System.out.println("-----------------------------------");
```
can make this line a void function so that the code looks less messy as this line appears many times

ZhangWenyue3325 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+        System.out.println("-----------------------------------");
+        for (int i = 0; i < tasksCount; i++){
+            System.out.print((i + 1) + ".");
+            System.out.print("[");
+            if (tasks[i].isDone){
+                System.out.print("X");
+            } else {
+                System.out.print(" ");
+            }
+            System.out.print("] ");
```
suggestions:
if (tasks[i].isDone){
    System.out.print("[X]");
} else {
    System.out.print("[ ]");
}

ZhangWenyue3325 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+            if (commandWords[0].equals("mark")) {
+                System.out.println("-----------------------------------");
+                System.out.println("Nice! I've marked this task as done:");
+                int index = Integer.parseInt(commandWords[1]);
+                System.out.println("[X] " + tasks[index - 1].name);
```
suggestions:
int index = Integer.parseInt(commandWords[1]) - 1;
then can use tasks[index] directly instead of tasks[index - 1] in the following lines

ZhangWenyue3325 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,20 @@
+import java.util.Scanner;
+
 public class Duke {
```
maybe can change the name Duke to Stella

ZhangWenyue3325 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+                System.out.println("-----------------------------------");
+                int index = Integer.parseInt(commandWords[1]);
+                System.out.println("[ ] " + tasks[index - 1].name);
+                tasks[index - 1].isDone = false;
+            } else {
+                Task.addTask(command);
+            }
+        } else {
+            Task.listTasks();
+        }
+    }
```
To many if else statements in this function, so the indentation looks messy
can consider use switch

ZhangWenyue3325 reviewed (commented):
Overall the code looks clean and neat. The commands are simple and rather short. However, the long functions such as void response are less structured. 
Good spacing!

eliztan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+            System.out.println(tasks[i].name);
+        }
+        System.out.println("-----------------------------------");
+    }
```
You could consider adding a empty line in between methods

eliztan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+        isDone = false;
+    }
+
+    public static void listTasks(){
```
Good job following camel case for method names

eliztan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+        System.out.println("-----------------------------------");
+    }
+
+    public static void response(String command){
```
"command" is a slightly ambiguous name especially as it is a string of multiple words, maybe consider changing it to a clearer name

eliztan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,60 @@
+                System.out.println("[X] " + tasks[index - 1].name);
+                System.out.println("-----------------------------------");
+                tasks[index - 1].isDone = true;
+            } else if (commandWords[0].equals("unmark")){
```
You could consider adding a space in between the ) and { at the end

PDHung1104 committed:
Add Parser class to classify task types

PDHung1104 committed:
- Implement UI class to handle greetings and user's input interface
- Implement enumerator TaskType to differentiate between Deadline, Todo, Event and Invalid task types.
- Refactor to have code blocks with same level of abstraction

PDHung1104 committed:
added exception handling methods

PDHung1104 committed:
Add delete task function.

wenxin-c commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,156 @@
+        if (commandWords.length != 2) {
+            throw new WrongInputFormat();
+        } else {
+            if (commandWords[0].length() < 10) {
```
Try not to use magic literals e.g. magic numbers and magic string, consider creating a constant and use the constant in your code instead.

wenxin-c commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,156 @@
+            System.out.println(UI.LINE_SEPARATOR);
+            return false;
+        } catch (MissingTaskName mtn) {
+            System.out.println(UI.LINE_SEPARATOR);
+            System.out.println("Missing task name");
+            System.out.println(UI.LINE_SEPARATOR);
```
Can consider abstracting this further to a printing method that takes in a string argument to avoid repeating yourself.

wenxin-c commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,127 @@
+    public static void addTask (Task task) {
+        System.out.println(UI.LINE_SEPARATOR);
+
+        switch (task.getTaskType()) {
+            case DEADLINE:
+            case EVENT:
+            case TODO:
+                System.out.println("Got it. I've added this task:");
+                tasks.add(task);
+                break;
+        }
```
Do follow the coding standard for switch block on our module website, i.e. ```switch``` and ```case``` should be aligned, there's no extra indentation in front of ```case```.

wenxin-c commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,127 @@
+    }
+
+    private static boolean isValidCommand (String command) {
+        return command.equals("list") || command.equals("mark") || command.equals("unmark") || command.equals("delete")
+                || isValidTask(command);
```
This is expression is very long and complex, try to simplify by breaking it down further.

wenxin-c reviewed (commented):
Well done in your ip so far. Try to be more careful of minor details and I would recommend to abstract your code further, especially the printing part to avoid repeating yourself.

PDHung1104 committed:
Add print message method, add save after changes

PDHung1104 committed:
Refactor code to be more OOP

PDHung1104 committed:
Add find function to search for tasks by keywords.

PDHung1104 committed:
Add JavaDoc

PDHung1104 committed:
Merge pull request #1 from PDHung1104/branch-Level-9

Add find function to search for tasks by keywords.

PDHung1104 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/MissingEntries.java
#	src/main/java/MissingTaskName.java
#	src/main/java/Parser.class
#	src/main/java/Parser.java
#	src/main/java/RedundantEntries.java
#	src/main/java/Storage.java
#	src/main/java/Task.class
#	src/main/java/Task.java
#	src/main/java/TaskList$1.class
#	src/main/java/TaskList.class
#	src/main/java/TaskList.java
#	src/main/java/UI.java
#	src/main/java/WrongInputFormat.java

PDHung1104 committed:
no message

PDHung1104 committed:
Merge pull request #3 from PDHung1104/fix-Storage-class

Fix Storage load method

PDHung1104 committed:
Add detailed UG and some fixes to the saving mechanism in the Storage class

PDHung1104 committed:
Merge branch 'UG'

# Conflicts:
#	src/main/java/Parser.class
#	src/main/java/Storage.class
#	src/main/java/Storage.java

PDHung1104 committed:
Add chatbot name to the top of the UG

PDHung1104 committed:
Merge pull request #6 from PDHung1104/UG

Add chatbot name to the top of the UG

PDHung1104 committed:
Update find command in the UG

PDHung1104 committed:
Merge pull request #7 from PDHung1104/UG

Add find command to UG

PDHung1104 committed:
Fix formatting problem in the UG

PDHung1104 committed:
Merge pull request #8 from PDHung1104/UG

Fix formatting problem in the UG
